### PR TITLE
Create policies for admin controllers

### PR DESCRIPTION
**Pull request by DenisRem**
[Trello Task](https://trello.com/_____)

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [ ] screenshot of feature is added if task has UI

### CODE REVIEW

**Review by mmarusyk — COMMENTED** [review]
Add, pls, tests for methods: `create?`, `new?`, `edit?`, `destroy?(where @current_user.admin?)`

File: app/policies/admin_policy.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @current_user = current_user
+    @user = model
+  end
+
+  def index?
+    @current_user.admin?
+  end
+
+  def show?
+    @current_user.admin? || @current_user == @user
```
Comment (pereslop) [inline app/policies/admin_policy.rb:16]: index? || @current_user == @user

File: app/policies/admin_policy.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  def initialize(current_user, model)
+    raise Pundit::NotAuthorizedError, "must be logged in" unless current_user
+
+    @current_user = current_user
+    @user = model
+  end
+
+  def index?
+    @current_user.admin?
```
Comment (pereslop) [inline app/policies/admin_policy.rb:12]: actually better to initialise instance varaible, something like user_admin = @current_user.admin? and reuse it in all your methods

**Review by matsevich — APPROVED** [review]

**Review by mmarusyk — APPROVED** [review]

File: app/policies/billing_contract_policy.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+class BillingContractPolicy < AdminPolicy
+  def index?
```
Comment (vasya539) [inline app/policies/billing_contract_policy.rb:2]: дуже багато однакового коду, якщо ці методи більше не будуть екстендитись то використай define_method

File: spec/controllers/account/admin/payments_controller_spec.rb [hunk 4]
```diff
@@ -10,13 +11,15 @@
   let!(:incorrect_type_of_file) { { file: fixture_file_upload('files/not_csv_file', 'text') } }
 
   describe 'GET#new_import' do
+    login_admin
```
Comment (chu4) [inline spec/controllers/account/admin/payments_controller_spec.rb:14]: you it on top so need for an extra one

File: spec/controllers/account/admin/payments_controller_spec.rb [hunk 5]
```diff
@@ -2,6 +2,7 @@
 
 RSpec.describe Account::Admin::PaymentsController, type: :controller do
   login_user
```
Comment (chu4) [inline spec/controllers/account/admin/payments_controller_spec.rb:4]: no need to login user as you login admin under it

File: spec/controllers/account/admin/users_controller_spec.rb [hunk 6]
```diff
@@ -14,6 +15,7 @@
   end
 
   describe 'DELETE#destroy' do
+    login_admin
```
Comment (chu4) [inline spec/controllers/account/admin/users_controller_spec.rb:18]: redundant you have line 6

File: spec/controllers/account/admin/osbbs_controller_spec.rb [hunk 7]
```diff
@@ -8,8 +8,10 @@
   let!(:invalid_params) { { name: '' } }
 
   login_user
```
Comment (chu4) [inline spec/controllers/account/admin/osbbs_controller_spec.rb:10]: redundant

File: spec/controllers/account/admin/osbbs_controller_spec.rb [hunk 8]
```diff
@@ -8,8 +8,10 @@
...
+  login_admin
 
   describe 'GET#index' do
+    login_admin
```
Comment (chu4) [inline spec/controllers/account/admin/osbbs_controller_spec.rb:14]: line 11

File: app/policies/billing_contract_policy.rb (hunk 3 above)
Comment (Axedez) [reply app/policies/billing_contract_policy.rb:2]: Вони будуть екстендитись!

Comment (Axedez) [reply app/policies/billing_contract_policy.rb:2]: В документації ці екшини описувались в application_controller, а потім при наслідуванні просто переписувались в дочірньому класі (якщо потрібно)

**Review by chu4 — APPROVED** [review]

**Pull request merged**